src/context/reducer.js

```diff
@@ -0,0 +1,43 @@
+export default function cartReducer(state, action) {
+    switch (action.type) {
+        case "ADD_TO_CART":
+            return {
+                ...state, cartItems: [...state.cartItems, action.payload]
+            }
+        case "INCREMENT_ITEMS":
+            return {
+                ...state, cartItems: state.cartItems.map(item => item.id === action.payload.id ? {
+                    ...item,
+                    quantity: item.quantity + 1
+                } : item)
+            }
+        case "DECREMENT_ITEMS":
+            if (action.payload.quantity === 1) {
+                return { ...state, cartItems: state.cartItems.filter(cartItem => cartItem.id !== action.payload.id) }
+            } return {
+                ...state, cartItems: state.cartItems.map(item => item.id === action.payload.id ? {
+                    ...item,
+                    quantity: item.quantity - 1
+                } : item)
+            }
+        case "REMOVE_FROM_CART":
+            return { ...state, cartItems: state.cartItems.filter(item => item.id !== action.payload.id) }
+
+        /** wishlist functions*/
+        case "MOVE_TO_CART":
+            if (state.cartItems.findIndex(item => item.id === action.payload.id) === -1) {
+                return { ...state, cartItems: [...state.cartItems, action.payload], wishList: state.wishList.filter(item => item.id !== action.payload.id) }
+            } return { ...state, wishList: state.wishList.filter(item => item.id !== action.payload.id) }
```

Comment:
I'm able to move stuff from wishlist to cart even if product is out of stock

Reply:
done